Tasks/Common/PowershellHelpers/Invoke-ActionWithRetries.ps1

```diff
@@ -0,0 +1,93 @@
+function Invoke-ActionWithRetries {
+    [CmdletBinding()]
+    param(
+        [scriptblock]
+        $Action,
+
+        [scriptblock]
+        $ActionSuccessValidator = { $true },
+
+        [int32]
+        $MaxRetries = 10,
+
+        [int32]
+        $RetryIntervalInSeconds = 1,
+
+        [ValidateScript({[System.Exception].IsAssignableFrom([type]$_)})]
+        $RetryableException,
+
+        [switch]
+        $ContinueOnError = $false,
+
+        [string]
+        $RetryMessage
+    )
+
+    Trace-VstsEnteringInvocation $MyInvocation
+
+    if($MaxRetries -eq 0)
+    {
+        $MaxRetries = [Int16]::MaxValue
+    }
+
+    if(!$RetryMessage)
+    {
+        $RetryMessage = Get-VstsLocString -Key RetryAfterMessage $RetryIntervalInSeconds
+    }
+
+    $retryIteration = 1
+    do 
+    {
+        $shouldRetry = $false
+        $result = $false
+        $exception = $null
+
+        try 
+        {
+            $result = & $Action
+        }
+        catch 
+        {
+            if(!$RetryableException -or ($_.Exception.GetType() -eq ([type]$RetryableException)))
```

Comment:
Consider: Rather than doing an exact type match check if the exception is derived from the RetryableException type (more in line with the behavior of catch)